core/whistle_services-1.0.0/src/whs_account_conversion.erl

```diff
@@ -0,0 +1,174 @@
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Set the reseller_id to the provided value on all the sub-accounts
+%% of the provided account
+%% @end
+%%--------------------------------------------------------------------
+-spec cascade_reseller_id(ne_binary(), ne_binary()) -> {'error', _} | 'ok'.
+cascade_reseller_id(Reseller, Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    ResellerId = wh_util:format_account_id(Reseller, 'raw'),
+    ViewOptions = [{<<"startkey">>, [AccountId]}
+                   ,{<<"endkey">>, [AccountId, wh_json:new()]}
+                  ],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, <<"accounts/listing_by_descendants">>, ViewOptions) of
+        {'error', _R}=Error ->
+            lager:debug("unable to determin descendants of ~s: ~p", [AccountId, _R]),
+            Error;
+        {'ok', JObjs} ->
+            _ = [set_reseller_id(ResellerId, SubAccountId)
+                 || JObj <- JObjs
+                        ,(SubAccountId = wh_doc:id(JObj)) =/= AccountId
+                ],
+            'ok'
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Set teh reseller_id to the provided value on the provided account
+%% @end
+%%--------------------------------------------------------------------
+-spec set_reseller_id(ne_binary(), ne_binary()) -> {'error', _} | 'ok'.
+set_reseller_id(Reseller, Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    ResellerId = wh_util:format_account_id(Reseller, 'raw'),
+    io:format("setting account ~s reseller id to ~s~n", [AccountId, ResellerId]),
+    _ = update_account_definition(AccountId, <<"pvt_reseller_id">>, ResellerId),
+    maybe_update_services(AccountId, <<"pvt_reseller_id">>, ResellerId).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_update_services(ne_binary(), ne_binary(), any()) -> {'error', _} | 'ok'.
+maybe_update_services(AccountId, Key, Value) ->
+    case couch_mgr:open_doc(?WH_SERVICES_DB, AccountId) of
+        {'error', _R}=Error ->
+            io:format("unable to open services doc ~s: ~p~n", [AccountId, _R]),
+            Error;
+        {'ok', JObj} ->
+            case couch_mgr:save_doc(?WH_SERVICES_DB, wh_json:set_value(Key, Value, JObj)) of
+                {'ok', _} -> 'ok';
+                {'error', _R}=Error ->
+                    io:format("unable to set ~s on services doc ~s: ~p~n", [Key, AccountId, _R]),
+                    Error
+            end
+    end.
+
+-spec update_account_definition(ne_binary(), ne_binary(), any()) -> {'error', _} | 'ok'.
+update_account_definition(AccountId, Key, Value) ->
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:open_doc(AccountDb, AccountId) of
```

Comment:
There is a `wh_util:account_update/1` which takes the modified account object and saves it to both the account db and the accounts db